File: objective-c/nilakantha.m
#include "nilakantha.h"

@implementation NilakanthaSeries

- (id)init {
  self = [super init];
  if (self) {
    a = 2L;
    s = 1L;
    pi = 3L;
  }
  return self;
}
- (void)iterate {
  pi += s * (4L / (a * (a * (a + 3L) + 2L)));
  a += 2;
  s = -s;
}
- (double)getPi {
  return pi;
}
@end

File: csharp/redone/Nilakantha.cs
using System;

namespace PiApproximator {
    class Nilakantha {
        private double a;
        private int s;
        private double pi;

        public Nilakantha() {
            a = 2;
            s = 1;
            pi = 3;
        }

        public void iterate() {
            pi += s * ( 4 / (double) (a * (a * (a + 3) + 2)));
            a += 2;
            s = -s;
        }

        public double getPi() {
            return pi;
        }
    }
}
